src/request/sign-transaction/SignTransactionApi.js

```diff
@@ -23,14 +24,29 @@ class SignTransactionApi extends TopLevelApi {
         parsedRequest.recipientLabel = this.parseLabel(request.recipientLabel);
         parsedRequest.transaction = this.parseTransaction(request);
         parsedRequest.layout = this.parseLayout(request.layout);
-        if (parsedRequest.layout === SignTransactionApi.Layouts.CHECKOUT) {
+        if (parsedRequest.layout === SignTransactionApi.Layouts.CHECKOUT
+            && request.layout === SignTransactionApi.Layouts.CHECKOUT) {
             parsedRequest.shopOrigin = this.parseShopOrigin(request.shopOrigin);
             parsedRequest.shopLogoUrl = this.parseShopLogoUrl(request.shopLogoUrl);
             if (parsedRequest.shopLogoUrl && parsedRequest.shopLogoUrl.origin !== parsedRequest.shopOrigin) {
                 throw new Errors.InvalidRequestError('origin of shopLogoUrl must be same as referrer');
```

Comment:
Since the referrer can't be checked in the Keyguard, this error message is technically wrong. And I think this is also checked in the Hub already? Why not:

```suggestion
                 throw new Errors.InvalidRequestError('origin of shopLogoUrl must be same as shopOrigin');
```